tools/orca/src/models/read.rs

```diff
@@ -0,0 +1,93 @@
+use crate::model::{self, ActorModel, Transition, TransitionAction, TransitionResult};
+
+use crate::error::Error;
+use crate::run::EventRecord;
+use crate::state::*;
+use kanidm_client::KanidmClient;
+
+use async_trait::async_trait;
+
+use std::time::Duration;
+
+enum State {
+    Unauthenticated,
+    Authenticated,
+}
+
+pub struct ActorReader {
+    state: State,
+}
+
+impl ActorReader {
+    pub fn new() -> Self {
+        ActorReader {
+            state: State::Unauthenticated,
+        }
+    }
+}
+
+#[async_trait]
+impl ActorModel for ActorReader {
+    async fn transition(
+        &mut self,
+        client: &KanidmClient,
+        person: &Person,
+    ) -> Result<EventRecord, Error> {
+        let transition = self.next_transition();
+
+        if let Some(delay) = transition.delay {
+            tokio::time::sleep(delay).await;
+        }
+
+        // Once we get to here, we want the transition to go ahead.
+        let (result, event) = match transition.action {
+            TransitionAction::Login => model::login(client, person).await,
+            TransitionAction::Logout => model::logout(client, person).await,
+            TransitionAction::PrivilegeReauth
+            | TransitionAction::WriteAttributePersonMail
+            | TransitionAction::ReadSelfAccount
+            | TransitionAction::WriteSelfPassword => return Err(Error::InvalidState),
+            TransitionAction::ReadSelfMemberOf => {
+                model::person_get_self_memberof(client, person).await
+            }
+        }?;
+
+        self.next_state(transition.action, result);
+
+        Ok(event)
+    }
+}
+
+impl ActorReader {
+    fn next_transition(&mut self) -> Transition {
+        match self.state {
+            State::Unauthenticated => Transition {
+                delay: Some(Duration::from_secs(2)),
+                action: TransitionAction::Login,
+            },
+            State::Authenticated => Transition {
+                delay: Some(Duration::from_secs(3)),
```

Comment:
Your problem here is that by having this at 3 seconds you create "bursts" where every actor delays and then reads then stops then reads. That's why your numbers are so chaotic.

Reply:
Yeah I agree introducing some randomness here will help. I've done it by passing the seed inside `as_dyn_object`, this way I don't need to do something like `Model::Writer(u64)` and `Model::Reader(u64)`, and as such I can use the `Model` enum as a ClapEnum, which looks really clean to me :D. No weird string conversions, only strong typing!